src/util-decode-mime.c

```diff
@@ -687,7 +687,11 @@ static uint8_t * GetFullValue(DataValue *dv, uint32_t *len)
 static inline uint8_t * FindBuffer(const uint8_t *src, uint32_t len, const uint8_t *find, uint32_t find_len)
 {
     /* Use utility search function */
-    return BasicSearchNocase(src, len, find, find_len);
+    if (find_len > UINT16_MAX) {
+        // TODO rewrite all this to have app-layer events
+        find_len = UINT16_MAX;
+    }
+    return BasicSearchNocase(src, len, find, (uint16_t)find_len);
```

Comment:
I wonder if we should just update `BasicSearchNocase` to accept `uint32_t`?

Reply:
`len` is `uint32_t`(haystack)
The needle is `uint16_t`
That looks reasonable...

Reply:
ok, makes sense. Can you add a ticket for tracking the events setting?

Reply:
Or should this mime code get rustified first ?

Reply:
For master that would be nice. Would we need a backport for this int handling though?

Reply:
I do not know.
It will be a big work I guess...
I am not sure it is worth it...